azure/templates/post-deploy.yml

```diff
@@ -20,81 +20,68 @@ steps:
-
-          make init
-          make apply aws_account_no=${aws_account_no} environment=$(WORKSPACE) \
-            client_id=$(INTROSPECTION_CLIENT_ID_INT) \
-            client_secret=$(INTROSPECTION_CLIENT_SECRET_INT) \
-            keycloak_environment=$(APIGEE_ENVIRONMENT)
-      elif ! [[ $APIGEE_ENVIRONMENT =~ .*-*sandbox ]]; then
-          export AWS_PROFILE=apim-dev
-          aws_account_no="$(aws sts get-caller-identity --query Account --output text)"
-
-          make init
-          make apply aws_account_no=${aws_account_no} environment=$(WORKSPACE) \
-            client_id=$(INTROSPECTION_CLIENT_ID) \
-            client_secret=$(INTROSPECTION_CLIENT_SECRET) \
-            keycloak_environment=$(APIGEE_ENVIRONMENT)
-      fi
-    displayName: Run Terraform
-    workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)/terraform"
-    retryCountOnTaskFailure: 2
-
   - bash: |
       set -e
       if ! [[ $APIGEE_ENVIRONMENT =~ .*-*sandbox ]]; then
-          if [ $(IS_IT_PR) -eq 1 ]; then
-              #if its not sandbox and it should be PR
-              source ../.venv/bin/activate
-              domain_name=https://$(make -s output name=service_domain_name)
-              oauth_endpoint="https://identity.ptl.api.platform.nhs.uk/auth/realms/gpconnect-pfs-mock-$APIGEE_ENVIRONMENT/protocol/openid-connect/token"
-              python ../scripts/apigee_kvm.py --env $APIGEE_ENVIRONMENT --access-token $(secret.AccessToken) populate-interaction-ids gp-connect-access-record-endpoints-pr-$(PR_NO) --ods REPC --provider-endpoint $domain_name --oauth-endpoint $oauth_endpoint
-          fi
+        export AWS_PROFILE=apim-dev
+        aws_account_no="$(aws sts get-caller-identity --query Account --output text)"
+      
+        service_name=$(FULLY_QUALIFIED_SERVICE_NAME)
+      
+        introspection_client_id=$(INTROSPECTION_CLIENT_ID)
+        introspection_client_secret=$(INTROSPECTION_CLIENT_SECRET)
+        if [[ $APIGEE_ENVIRONMENT = int ]]; then
+          introspection_client_id=$(INTROSPECTION_CLIENT_ID_INT)
+          introspection_client_secret=$(INTROSPECTION_CLIENT_SECRET_INT)
+        fi
+        
+        pr_no=$(echo $service_name | grep -oE '[0-9]+$')
+        if [ -z $pr_no ]; then
+          workspace=$APIGEE_ENVIRONMENT
+        else
+          workspace=pr-$pr_no
+        fi
+       
+        echo Deploy mock provider with following parameters:
+        echo service_name: $service_name
+        echo workspace: $workspace
+        echo Apigee environment: $APIGEE_ENVIRONMENT
+        echo pr_no: $pr_no
+        echo introspection client id: $introspection_client_id
+        
+        cd terraform
+        make init
+        make apply aws_account_no=${aws_account_no} environment=$workspace \
+          client_id=$introspection_client_id \
+          client_secret=$introspection_client_secret \
+          keycloak_environment=$APIGEE_ENVIRONMENT
+        cd ..
```

Comment:
`make apply` was the last command of bash before, when it fails, step fails and exit, for that there is retry option, so it automatically retries itself. But here as there are more commands after make apply, this will not exit here, so pipeline will fail for first time. We have to run it manually then